tests/PSRule.Rules.Azure.Tests/Azure.VNET.Tests.ps1

```diff
@@ -209,6 +212,58 @@ Describe 'Azure.VNET' -Tag 'Network', 'VNET' {
             $ruleResult.Length | Should -Be 2;
             $ruleResult.TargetName | Should -BeIn 'vnet-F', 'vnet-G';
         }
+
+        It 'Azure.VNET.PrivateSubnet' {
+            $filteredResult = $result | Where-Object { $_.RuleName -eq 'Azure.VNET.PrivateSubnet' };
+
+            # Fail
+            $ruleResult = @($filteredResult | Where-Object { $_.Outcome -eq 'Fail' });
+            $ruleResult.Length | Should -Be 7;
+            $ruleResult.TargetName | Should -Be 'vnet-A', 'vnet-B', 'vnet-C', 'vnet-D', 'vnet-G', 'vnet-H/subnet-A', 'vnet-H/subnet-B';
+
+            $ruleResult[0].Reason | Should -BeExactly @(
+                "The subnet (GatewaySubnet) should disable default outbound access."
```

Comment:
The documentation is not clear on this but it wouldn't work for this special named subnet. The `GatewaySubnet` subnet should be excluded from the rule.